skorch/net.py

```diff
@@ -2095,11 +2098,13 @@ def _register_attribute(
             self.cuda_dependent_attributes_ = (
                 self.cuda_dependent_attributes_[:] + [name + '_'])
 
-        if self.init_context_ == 'module':
+        # make sure to not double register -- this should never happen, but
+        # still better to check
+        if (self.init_context_ == 'module') and (name not in self._modules):
             self._modules = self._modules[:] + [name]
```

Comment:
I think this bound is what fixed the bug. Would a `set` be a better data structure for `_modules`?